tests/test_db_models.py

```diff
@@ -0,0 +1,94 @@
+from unittest import TestCase
+from app import create_app
+from models import db, Order, Purchase, Product, Subproduct, Seller
+import datetime
+
+class ModelsTestCase(TestCase):
+    def setUp(self):
+        # create the app testing environment
+        create_app('TESTING')
+
+        # delete and create tables
+        db.drop_all()
+        db.create_all()
+
+        # add a seller row
+        self.EMAIL = "KidsKrafts4U@gmail.com"
+        new_seller = Seller(email = self.EMAIL, password = '1234')
+        db.session.add(new_seller)
+        db.session.commit()
+
+        # add a product row and subproducts for that product
+        subproducts = [{'name': "Punxsutawney Phil", 'image_url': "../static/seed_images/kit_1/punxsutawney_phil.jpg"}, {'name': "Snowstorm Painting", 'image_url': "../static/seed_images/kit_1/snowstorm_painting.jpg"}, {'name': "valentine_pup", 'image_url': "../static/seed_images/kit_1/valentine_pup.jpg"}, {'name': "Marshmallow Snowman", 'image_url': "../static/seed_images/kit_1/marshmallow_snowman.jpg"}]
+        new_product = Product(name = "February Kit 21", price = 5.00, image_url = "../static/seed_images/kit_1/product_image.jpg", category = "not_selling")
+        db.session.add(new_product)
+        db.session.commit()
+        for subproduct in subproducts:
+            new_subproduct = Subproduct(product_id = new_product.id, name = subproduct["name"], image_url = subproduct["image_url"])
+            db.session.add(new_subproduct)
+            db.session.commit()
```

Comment:
Try to avoid any kind of conditional statements or loops in test code. You want your tests to be as simple as possible. Refactor accordingly.